loglist2/logfilter.go

```diff
@@ -24,6 +24,14 @@ import (
 // root-certificates.
 type LogRoots map[string]*ctfe.PEMCertPool
 
+// Compatible creates a new LogList containing only Logs matching the temporal,
+// root-acceptance and Log-status conditions.
+func (ll *LogList) Compatible(cert *x509.Certificate, certRoot *x509.Certificate, roots LogRoots, lstat LogStatus) LogList {
+	active := ll.TemporalCompatible(cert)
```

Comment:
Could this be:

```golang
return ll.TemporalCompatible(cert)
             .RootCompatible(certRoot, roots)
             .SelectyByStatus(lstat)
```

?

Reply:
Would be nice. But those methods operate on pointers and return objects. And return values are not so easily addressable. Probably there's an elegant solution I miss.

Reply:
Could the filtering methods all return `*LogList` rather than `LogList`?

Reply:
They could but why? They produce new instances.

Reply:
Well, if nothing else it'd bring them into line with what the c'tors for creating new `LogList`s return - it looks like these filter methods are unusual in that they return value types.

As a side-effect that'd also let you do the method chaining, if you decided that was a good idea.